FWCore/Services/plugins/DependencyGraph.cc

```diff
@@ -398,7 +373,7 @@ void DependencyGraph::lookupInitializationComplete(PathsAndConsumesOfModulesBase
   }
 
   // remove the nodes corresponding to the modules that have been removed from the process
-  for (int i = boost::num_vertices(m_graph) - 1; i > 1; --i) {
+  for (int i = boost::num_vertices(m_graph) - 1; i > 0; --i) {
```

Comment:
IIRC `0` should be the `source`, so it cannot be removed ?

Reply:
I don't think we guarantee `source` to have module ID 0, i.e. if that happens today, it is accidental and may change in the future. Therefore looping over all nodes without assumptions would sound most future-proof to me.

Reply:
With the newest version I just pushed this part of the code is deleted.